freeipa/src/main/resources/schema/app/20190516151400_add_tenant_to_stack.sql

```diff
@@ -0,0 +1,32 @@
+-- // Create Changelog
+
+-- Default DDL for changelog table that will keep
+-- a record of the migrations that have been run.
+
+-- You can modify this to suit your database before
+-- running your first migration.
+
+-- Be sure that ID and DESCRIPTION fields exist in
+-- BigInteger and String compatible fields respectively.
+
+ALTER TABLE stack
+    ADD tenant VARCHAR(255);
+
+CREATE UNIQUE INDEX tenant_uindex
+    ON stack (tenant);
+
+CREATE UNIQUE INDEX environment_tenant_uindex
+    ON stack (name, environment, tenant);
```

Comment:
what about the environment_tenant_uindex. should we enforce the 1-1 environment to stack relationship? Or you could remove the "UNIQUE" if using this for performance.

if we keep this index, remove the "name" field from the index.